src/Http/Controllers/FallbackController.php

```diff
@@ -2,32 +2,58 @@
 
 namespace Rapidez\Core\Http\Controllers;
 
+use Exception;
 use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Database\RecordsNotFoundException;
+use Illuminate\Http\Request;
 use Illuminate\Routing\Exceptions\BackedEnumCaseNotFoundException;
 use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Cache;
 use Rapidez\Core\Facades\Rapidez;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 use Symfony\Component\Routing\Exception\RouteNotFoundException;
 
 class FallbackController
 {
-    public function __invoke()
+    public function __invoke(Request $request)
     {
-        foreach (Rapidez::getAllFallbackRoutes() as $route) {
-            try {
-                $response = App::call($route['action']['uses']);
+        $cacheKey = 'fallbackroute-' . md5($request->url());
+        $route = Cache::get($cacheKey);
+        if($route && $response = $this->tryRoute($route)) {
+            return $response;
+        }
 
-                // Null response is equal to no response or 404.
-                if ($response === null) {
-                    abort(404);
-                }
+        foreach (Rapidez::getAllFallbackRoutes() as $route) {
+            if(!($response = $this->tryRoute($route))) {
+                continue;
+            }
 
-                return $response;
-            } catch (RouteNotFoundException|NotFoundHttpException|BackedEnumCaseNotFoundException|ModelNotFoundException|RecordsNotFoundException $e) {
+            try {
+                Cache::put($cacheKey, $route, value(config('rapidez.fallback_routes.cache_duration', 3600), $request, $response, $route));
+            } catch (Exception $e) {
+                // We can't cache it, no worries.
+                // Ususally a sign it's a direct callback or caching hasn't been configured properly.
```

Comment:
Why do we want to catch this?

Reply:
Because anything to do with saving the controller in cache shouldn't hinder returning a response.

Cases that i've had error pages on this were because of anonymous crontrollers `Rapidez::addFallbackRoute(fn() => 'test')` because an anonymous function can not be serialized for storage in cache.

And an improperly configured redis application, e.g. driver missing, redis full

Caching the controller for a request is not necessary but can speed up the application when one in the list has low speed

Reply:
But Laravels route caching does have support for anonymous callback functions? Catching for misconfiguration isn't needed as caching is used on multiple places where we don't try/catch.

Reply:
That's true Laravel's route caching does indeed work with anonymous callback functions because it caches the routes per-route in a "routes.php" file while we're caching it in Redis per-url.

Redis can only save strings, so we need to Serialize the functions. In the case of referenced functions this is easy as it is just pointing to the function that should be executed.

When it is a direct function however you will get the following error:
![image](https://github.com/rapidez/core/assets/15870933/f7af250d-69c4-47b2-b6dd-b462eaf1fefe)
As transforming a function into a string is not possible by default

There may be ways to hack it by working with exec with the string stored in cache, but that can open up many other issues